templates/test/ci/prow-ci-version-gmsa/patches/machine-identity.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-md-win
+spec:
+  template:
+    spec:
+      identity: UserAssigned
+      userAssignedIdentities:
+      - providerID: "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${CI_RG}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/cloud-provider-user-identity"
```

Comment:
doesn't this require the fix in #2214 to work?

Reply:
I had similar question but after discussing with @mboersma we figured out that since we are not testing any cloudprovider specific flows in this set of tests it doesn't exercise the code that would cause issues.